### PR TITLE
Implement refreshtoken

### PR DESCRIPTION
**Pull request by Shavronskyy**
#187

### CODE REVIEW

File: Streetcode/Streetcode.BLL/DTO/Users/RefreshTokenDTO.cs [hunk 1]
```diff
@@ -8,6 +8,8 @@ namespace Streetcode.BLL.DTO.Users
 {
     public class RefreshTokenDTO
     {
-        public string Token { get; set; } = string.Empty;
+        public string Token { get; set; }
+        public DateTime Created { get; set; } = DateTime.Now;
+        public DateTime Expires { get; set; }
```
Comment (serhiiyasenev) [inline Streetcode/Streetcode.BLL/DTO/Users/RefreshTokenDTO.cs:13]:
```
public DateTime Created { get; set; } = DateTime.Now;
public DateTime Expires { get; set; }
```

it should be encapsulated inside `RefreshToken` string

File: Streetcode/Streetcode.BLL/DTO/Users/TokenResponseDTO.cs [hunk 2]
```diff
@@ -6,9 +6,9 @@
 
 namespace Streetcode.BLL.DTO.Users
 {
-    public class RefreshTokenResponce
+    public class TokenResponseDTO
     {
-        public string Token { get; set; } = string.Empty;
-        public DateTime ExpireAt { get; set; }
+        public RefreshTokenDTO RefreshToken { get; set; }
```
Comment (serhiiyasenev) [inline Streetcode/Streetcode.BLL/DTO/Users/TokenResponseDTO.cs:11]: we don't need it - `RefreshToken` it's just a string

File: Streetcode/Streetcode.BLL/MediatR/Account/RefreshToken/RefreshTokenHandler.cs [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        public RefreshTokensHandler(UserManager<User> userManager, ILoggerService logger, AccessTokenConfiguration accessTokenConfiguration)
+        {
+            _logger = logger;
+            _userManager = userManager;
+            _accessTokenConfiguration = accessTokenConfiguration;
+        }
+
+        public async Task<Result<User>> Handle(RefreshTokensCommand request, CancellationToken cancellationToken)
+        {
+            _tokenService = new(_userManager, _accessTokenConfiguration, _logger);
```
Comment (serhiiyasenev) [inline Streetcode/Streetcode.BLL/MediatR/Account/RefreshToken/RefreshTokenHandler.cs:29]: `_tokenService` should be created via constructor

Comment (Shavronskyy) [reply Streetcode/Streetcode.BLL/MediatR/Account/RefreshToken/RefreshTokenHandler.cs]: resolved

File: Streetcode/Streetcode.BLL/DTO/Users/TokenResponseDTO.cs (hunk 2 above)
Comment (Shavronskyy) [reply Streetcode/Streetcode.BLL/DTO/Users/TokenResponseDTO.cs:11]: refresh token have created and expiration time

Comment (Shavronskyy) [reply Streetcode/Streetcode.BLL/DTO/Users/TokenResponseDTO.cs:11]: and refreshtokendto was already in the project, I only add my prop

File: Streetcode/Streetcode.DAL/Persistence/StreetcodeDbContext.cs [hunk 4]
```diff
@@ -84,7 +84,7 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
 
     //protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
     //{
-    //    optionsBuilder.UseSqlServer("Data Source=(localdb)\\mssqllocaldb;Initial Catalog=StreetcodeDB_Mig_Target;Integrated Security=True");
+    //    optionsBuilder.UseSqlServer("Data Source=(localdb)\\MSSQLLocalDB;Initial Catalog=master;Integrated Security=True");
 
     //    base.OnConfiguring(optionsBuilder);
     //}
```
Comment (metrondir) [inline Streetcode/Streetcode.DAL/Persistence/StreetcodeDbContext.cs:90]: erase that

Comment (Shavronskyy) [reply Streetcode/Streetcode.DAL/Persistence/StreetcodeDbContext.cs]: done

File: Streetcode/Streetcode.BLL/DTO/Users/RefreshTokenDTO.cs (hunk 1 above)
Comment (Shavronskyy) [reply Streetcode/Streetcode.BLL/DTO/Users/RefreshTokenDTO.cs:13]: how?
refreshtoken string its just random 64 symbols

Comment (metrondir) [reply Streetcode/Streetcode.BLL/DTO/Users/RefreshTokenDTO.cs:13]: we decided not to insert a  Expires into the refresh . its just a random string

**Review by metrondir — APPROVED** [review]
lgtm

File: Streetcode/Streetcode.BLL/MediatR/Account/RefreshToken/RefreshTokenHandler.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+        private TokenService _tokenService;
+        private UserManager<User> _userManager;
+        private AccessTokenConfiguration _accessTokenConfiguration;
+
+        public RefreshTokensHandler(UserManager<User> userManager, ILoggerService logger, AccessTokenConfiguration accessTokenConfiguration)
+        {
+            _logger = logger;
+            _userManager = userManager;
+            _accessTokenConfiguration = accessTokenConfiguration;
+            _tokenService = new(_userManager, _accessTokenConfiguration, _logger);
```
Comment (serhiiyasenev) [inline Streetcode/Streetcode.BLL/MediatR/Account/RefreshToken/RefreshTokenHandler.cs:25]: `_tokenService = tokenService;`

Comment (Shavronskyy) [reply Streetcode/Streetcode.BLL/MediatR/Account/RefreshToken/RefreshTokenHandler.cs:25]: I tried, it didn't work like this

**Review by BohdanLytvynov — APPROVED** [review]

**Pull request merged**